src/openMVG/robust_estimation/rand_sampling.hpp

```diff
@@ -35,10 +35,11 @@
 namespace openMVG {
 namespace robust{
 
+// rperrot : is this legal ? why using an anonymous namespace here ? 
 namespace
 {
   std::default_random_engine random_generator;
```

Comment:
It doesn't make sense to use anonymous namespaces in headers. It will duplicate the symbols in each cpp that include this header. If you really need a global variable you need to declare it in static in the cpp file and in extern in the header.

Here, you can declare it as a static variable directly inside the function.